Jenkinsfile

```diff
@@ -125,6 +121,94 @@ pipeline {
               '''
             }
           }
+
+          post {
+            always {
+              archiveArtifacts "results/**/*"
+              junit "results/results.xml"
+              withCredentials([
+                [
+                  $class: "AmazonWebServicesCredentialsBinding",
+                  credentialsId: "f40eebe0-f9aa-4336-b460-b2c4d7876fde",
+                  accessKeyVariable: "AWS_ACCESS_KEY_ID",
+                  secretKeyVariable: "AWS_SECRET_ACCESS_KEY"
+                ]
+              ]) {
+                sh "./system-tests/_scripts/delete-cluster.sh"
+              }
+            }
+          }
+        }
+      }
+    }
+
+    stage("System Test 1.13 nightly") {
+      when {
+        expression {
+          master_branches.contains(BRANCH_NAME)
+        }
+      }
+      steps {
+        withCredentials([
+          [
+            $class: "AmazonWebServicesCredentialsBinding",
+            credentialsId: "f40eebe0-f9aa-4336-b460-b2c4d7876fde",
+            accessKeyVariable: "AWS_ACCESS_KEY_ID",
+            secretKeyVariable: "AWS_SECRET_ACCESS_KEY"
+          ]
+        ]) {
+          sh '''
+            INSTALLER_URL="https://downloads.dcos.io/dcos/testing/1.13/dcos_generate_config.sh" ./system-tests/_scripts/launch-cluster.sh
+            export CLUSTER_URL=\$(cat /tmp/cluster_url.txt)
+            export CLUSTER_AUTH_TOKEN=\$(./system-tests/_scripts/get_cluster_auth.sh)
+            export CLUSTER_AUTH_INFO=\$(echo '{ "uid": "albert@bekstil.net", "description": "albert" }' | base64)
+            DCOS_CLUSTER_SETUP_ACS_TOKEN="\$CLUSTER_AUTH_TOKEN" dcos cluster setup "\$CLUSTER_URL" --provider=dcos-oidc-auth0 --insecure
+            npm run test:system
+          '''
+        }
+      }
+
+      post {
+        always {
+          archiveArtifacts "results/**/*"
+          junit "results/results.xml"
+          withCredentials([
+            [
+              $class: "AmazonWebServicesCredentialsBinding",
+              credentialsId: "f40eebe0-f9aa-4336-b460-b2c4d7876fde",
+              accessKeyVariable: "AWS_ACCESS_KEY_ID",
+              secretKeyVariable: "AWS_SECRET_ACCESS_KEY"
+            ]
+          ]) {
+            sh "./system-tests/_scripts/delete-cluster.sh"
+          }
+        }
+      }
+    }
+
+    stage("System Test 1.12 nightly") {
```

Comment:
+1

Also, I would like to ask if this might be the point in time where we switch from declarative builds to a scripted build, because of the complexity that our build has by now.

Reply:
splitting it up in multiple jobs is something I'd love to have, but failed a couple times trying to implement it in a short timeframe. I'm attacking this with iterative progress here. having them _at all_ as a first step enables us to catch issues - ha, _at all_, I'd love to see it improved later on.

Reply:
my current idea would be that we have all tests running in parallel on every PR again :) this would solve the issue 1 failing another and also would not add runtime at all like the current approach. having them in multiple jobs is a possible approach, as long as we're aware of the downsides (multiple builds) or can figure out a solution to pass the build into childjobs (thats where I failed)